### DISCUSSION TITLE
Custom Integration

### DISCUSSION THREAD

**Discussion by akelyasir** [leptos-rs/leptos#2691] [Q&A] [answered]
Hi,

I want to use Leptos SSR. But I don't want to use the leptos watch feature, server and cargo.toml features. I want to integrate it to my own server application. Is there a way to use SSR without the server feature? 

This may be an illogical question, I don't know much about Leptos. For example, SSR application can be done with dioxus virtual dom, this way it is easy to integrate it into special applications. I wanted to know if there is a similar way for Leptos.

Thanks

**Comment by gbj** [comment, marked as answer]
Sure. 

You don't want to use: 
- `cargo leptos watch`: not part of the crate, `cargo-leptos` is a (very convenient) separate build tool
- server: not sure what you mean by this
- Cargo.toml: I assume you're referring specifically to the `cargo-leptos` config sections of Cargo.toml, which of course you don't need if you're not using `cargo-leptos`

It is fairly straightforward to render a `View` directly to HTML (https://docs.rs/leptos/latest/leptos/ssr/fn.render_to_string.html). All the rest is just for convenience. 

If you try and run into more specific issues feel free to ask, of course.

**Reply by akelyasir** [reply]
Hi @gbj 

Thank you very much for your help. You got it right, thank you. Are there any disadvantages to using it this way? Will I have any problems with speed and usability? My purpose in using it is to be able to use it in a special backend structure. To use data from the backend as an html template (like tera) and to get the advantages of leptos.

**Comment by akelyasir** [comment]
Hi,

Is there any loss of performance when used this way? Also, can I still use all of Leptos' abilities?

Thanks

**Reply by gbj** [reply]
No and yes.
